source/base/tensor_product_polynomials.cc

```diff
@@ -598,6 +596,22 @@ TensorProductPolynomials<dim, PolynomialType>::evaluate(
 
 
 
+template <>
+void
+TensorProductPolynomials<0, Polynomials::Polynomial<double>>::evaluate(
+  const Point<0> &,
+  std::vector<double> &,
+  std::vector<Tensor<1, 0>> &,
+  std::vector<Tensor<2, 0>> &,
+  std::vector<Tensor<3, 0>> &,
+  std::vector<Tensor<4, 0>> &) const
+{
+  constexpr int dim = 0;
+  AssertThrow(dim > 0, ExcNotImplemented());
```

Comment:
Why write it like this, and not just as follows?
```suggestion
  AssertThrow(false, ExcNotImplemented());
```

Reply:
Good idea, I did this (and changed all other places in the file) in a separate commit.